Updated the installation guide using Ansible  for Navarino release (#272)
sodafoundation/documentation: content/soda-gettingstarted/installation-using-ansible.md

---
This document describes how to install SODA projects in a local cluster with detailed configuration options. These steps will help you to enable / disable projects. After installation using these steps, you can get the features of all the enabled projects. You can test either through SODA Dashboard UI or CLI

{{% notice info %}}

> **Below process of installation will support for Ubuntu 20.04 & Ubuntu 18.04.** <br />

Reviewer: _Below process of installation will support for Ubuntu 20.04 & Ubuntu 18.04_  -> **This installation guide is supported on Ubuntu 18.04 & Ubuntu 20.04.**

Author: updated

> **Below process of installation will support for Ubuntu 20.04 & Ubuntu 18.04.** <br />

Reviewer: _Below process of installation will support for Ubuntu 20.04 & Ubuntu 18.04_  -> **This installation guide is supported on Ubuntu 18.04 & Ubuntu 20.04.**

Author: updated

{{% /notice %}}

---

### Pre-requisites
- SODA installation is tested on `Ubuntu 18.04`& `Ubuntu 20.04`. 
- Python 3.6 or above should be installed

```bash
sudo apt-get update && sudo apt-get install -y git
```

> - Ensure no ansible & docker installed, OR Lastest ansible and docker tools are installed with versions listed below or later. If ansible & docker are not installed in the OS, script `install_dependencies.sh` will install it.  

Reviewer: _Please change this one line as below:_
- Ensure Ansible & Docker are not installed
- If Ansible and Docker are installed, ensure that the latest version of Ansible and Docker tools are installed with versions listed below.
- If Ansible & Docker are not installed then use the `install_dependencies.sh` script, as described below, to install them.

Author: Updated

> - Ensure no ansible & docker installed, OR Lastest ansible and docker tools are installed with versions listed below or later. If ansible & docker are not installed in the OS, script `install_dependencies.sh` will install it.  

Reviewer: Where are the Ansible and Docker versions listed?

Author: Updated

---
This document describes how to install SODA projects in a local cluster with detailed configuration options. These steps will help you to enable / disable projects. After installation using these steps, you can get the features of all the enabled projects. You can test either through SODA Dashboard UI or CLI

{{% notice info %}}
**Below process of installation will support for Ubuntu 20.04 & Ubuntu 18.04.** <br />
{{% /notice %}}

---

### Pre-requisites
- SODA installation is tested on `Ubuntu 18.04`& `Ubuntu 20.04`. 
- Python 3.6 or above should be installed

```bash
sudo apt-get update && sudo apt-get install -y git
```

> - Ensure no ansible & docker installed, OR Lastest ansible and docker tools are installed with versions listed below or later. If ansible & docker is not installed in the OS, script          install_dependencies.sh will install it.

Reviewer 2: ```suggestion
- Ensure no ansible & docker installed, OR Lastest ansible and docker tools are installed with versions listed below or later. If ansible & docker is not installed in the OS, script          `install_dependencies.sh` will install it.
```

Reviewer: If ansible & docker is not installed in the OS -> If ansible & docker **are** not installed in the OS

---
This document describes how to install SODA projects in a local cluster with detailed configuration options. These steps will help you to enable / disable projects. After installation using these steps, you can get the features of all the enabled projects. You can test either through SODA Dashboard UI or CLI

{{% notice info %}}
**This installation guide is supported on Ubuntu 18.04 & Ubuntu 20.04.** <br />
{{% /notice %}}

---

### Pre-requisites
- SODA installation is tested on `Ubuntu 18.04`& `Ubuntu 20.04`. 

```bash
sudo apt-get update && sudo apt-get install -y git
```

- Python 3.6 or above should be installed 
- If Ansible, Docker & Go are not installed in the OS, script `install_dependencies.sh` will install it.
- If Ansible, Docker & Go are installed ensure that the version as listed below.
  - Docker: Latest (Verfied on 20.10.21)

>   - Ansible: Latest (Verfied on 5.10.0 )

Reviewer: Please specify that this is the Ansible package version. Also mention the ansible-core version that is supported.

host_ip: 127.0.0.1
```

---

### Select SODA Projects to install

> To install SODA Projects and enable the different features, variables have to be modified in the respective files as below:

Reviewer 2: ```suggestion
To install SODA Projects and enable the different features, installer needs to be configured using the respective files as below:
```

> **_Note: Hotpot ,Gelato & Delfin can be used either through dashboard or REST APIs._**

Reviewer: Hotpot -> Terra (previously known as Hotpot)
Gelato -> Strato (SODA Multicloud project, previously known as Gelato)

---
#### Enable Delfin installation
delfin (Dolphin in Spanish!), the SODA Infrastructure Manager project is an open source project to provide unified, intelligent and scalable resource management, alert and performance monitoring. It covers the resource management of all the backends & other infrastructures under SODA deployment. It also provides alert management and metric data (performance/health) for monitoring and further analysis.

>   -Enable following configration  

Reviewer 2: ```suggestion
  -Enable following configuration  
```

    - In file installer/ansible/group_vars/delfin.yml `enable_delfin: true`  
    - In file installer/ansible/group_vars/srm-toolchain.yml `install_srm_toolchain: true`  
    - In file installer/ansible/group_vars/dashboard.yml `enable_dashboard: true`  

>     - **[Click here](#configure-delfin-installation) for more config**

Reviewer 2: ```suggestion
- **[Click here](#configure-delfin-installation) for other supported configuration details **
```

---

#### Enable Hotpot  
 Hotpot installs SODA On Premise only.  
- Enable following configurations
  - In file installer/ansible/hotpot.yml  update the value  `enable_hotpot : true`.
  - In file installer/ansible/group_vars/dashboard.yml `enable_dashboard: true`
  - In file installer/ansible/group_vars/common.yml `host_ip : <User's IP address, eg. 127.0.0.1>`  

>   - **[Click here](#configure-soda-on-premise-installation) for more config**  

Reviewer 2: Change as above

> #### Enable Gelato

Reviewer: Enable Strato.
let's use the project name Strato in the description. Mention the variable as gelato.

Gelato installs SODA Multicloud only.
- Enable following configurations  
  - In file installer/ansible/group_vars/gelato.yml  update the value  `enable_gelato : true`.
  - In file installer/ansible/group_vars/dashboard.yml `enable_dashboard: true`
  - In file installer/ansible/group_vars/common.yml `host_ip : <User's IP address, eg. 127.0.0.1>`  

>   - **[Click here](#enable-storage-service-plans-in-multicloud-\(optional\)) for more config** 

Reviewer 2: Change as above

---

> ### Install SODA

Reviewer 2: ```suggestion
### Run SODA Installer
```

- Enable following configurations  
  - In file installer/ansible/group_vars/delfin.yml `enable_delfin: true`  
  - In file installer/ansible/group_vars/srm-toolchain.yml `install_srm_toolchain: true`  
  - In file installer/ansible/group_vars/dashboard.yml `enable_dashboard: true`  
  - **[Click here](#configure-delfin-installation) for other supported configuration details**
---

#### Enable Terra  
 Terra installs SODA On Premise only.  
- Enable following configurations  
  - In file installer/ansible/hotpot.yml  update the value  `enable_hotpot : true`.
  - In file installer/ansible/group_vars/dashboard.yml `enable_dashboard: true`
  - In file installer/ansible/group_vars/common.yml `host_ip : <User's IP address, eg. 127.0.0.1>`  
  - **[Click here](#configure-soda-on-premise-installation) for other supported configuration details**   

> 

Reviewer: Add a line here ---

#### Enable Strato
Strato installs SODA Multicloud only.
- Enable following configurations  
  - In file installer/ansible/group_vars/gelato.yml  update the value  `enable_gelato : true`.
  - In file installer/ansible/group_vars/dashboard.yml `enable_dashboard: true`
  - In file installer/ansible/group_vars/common.yml `host_ip : <User's IP address, eg. 127.0.0.1>`  

>   - **[Click here](#enable-storage-service-plans-in-multicloud) other supported configuration details**

Reviewer: Click here **for** other.....

---
### Run SODA Installer
Run SODA installer ansible playbook to start the deployment

Check if the hosts can be reached

```bash  
sudo -E env "PATH=$PATH" ansible all -m ping -i local.hosts
```

```bash
sudo -E env "PATH=$PATH" ansible-playbook site.yml -i local.hosts
# You can use the -vvv or -vv option to enable verbose display and debug mode.
[verbosity level: -vv < -vvv]
sudo -E env "PATH=$PATH" ansible-playbook site.yml -i local.hosts -vvv
```

SODA ansible installer supports installation using tags. To install a particular service use the playbook as follows
```bash
# This installs only delfin

> sudo -E env "PATH=$PATH" ansible-playbook site.yml -i local.hosts -vvv --tags delfin

Reviewer: Please check if the `delfin` tag works. @joseph-v Does it work?

Reviewer 2: > Please check if the `delfin` tag works. @joseph-v Does it work?

Yes

--- 

### Configure SRM Toolchain installation
Delfin produces metrics which can be consumed by any of the exporters that are supported. Currently Delfin supports the Prometheus and Kafka exporters.
The SRM Toolchain is required to view the metrics and visualization in the SODA Dashboard.\  
Installing the SRM toolchain will install Prometheus, AlertManager and Grafana versions as per the configuration below and can be changed. 

> 

Reviewer: These instructions are missing which file needs to be modified. 
This was available earlier. Why is it removed?
_Update the file `ansible/group_vars/srm-toolchain.yml` and change the value of `install_srm_toolchain` to `true`.  
If this value is set to false then the metrics and visualization will not be available using SODA Dashboard._

Author: Previously two blocks were there for SRM toolchain 
1st for definition and 2nd for configuration 
now we refactored and combined in both in one block 
and install_srm_toolchain is available and in the Delfin installation process install_srm_toolchain to true is given

Reviewer: > Previously two blocks were there for SRM toolchain 1st for definition and 2nd for configuration now we refactored and combined in both in one block and install_srm_toolchain is available and in the Delfin installation process install_srm_toolchain to true is given

ok
